templates/python/custom-copilot-rag-azure-ai-search/README.md.tpl

```diff
@@ -0,0 +1,142 @@
+{{/useOpenAI}}
+> - An [Azure Search service](https://learn.microsoft.com/en-us/azure/search/search-what-is-azure-search).
+{{^enableTestToolByDefault}}
+> - A [Microsoft 365 account for development](https://docs.microsoft.com/microsoftteams/platform/toolkit/accounts).
+{{/enableTestToolByDefault}}
+{{#enableTestToolByDefault}}
+> - [Node.js](https://nodejs.org/) (supported versions: 16, 18) for local debug in Test Tool.
+{{/enableTestToolByDefault}}
+
+### Configurations
+1. First, Open the command box and enter `Python: Create Environment` to create and activate your desired virtual environment. Remember to select `requirements.txt` as dependencies to install when creating the virtual environment.
+{{#enableTestToolByDefault}}
+{{#useAzureOpenAI}}
+1. In file *env/.env.testtool.user*, fill in your Azure OpenAI key `SECRET_AZURE_OPENAI_API_KEY`, deployment name `AZURE_OPENAI_MODEL_DEPLOYMENT_NAME`, endpoint `AZURE_OPENAI_ENDPOINT` and embedding deployment name `AZURE_OPENAI_EMBEDDING_DEPLOYMENT`.
+{{/useAzureOpenAI}}
+{{#useOpenAI}}
+1. In file *env/.env.testtool.user*, fill in your OpenAI key `SECRET_OPENAI_API_KEY`. 
+1. In this template, default model name is `gpt-3.5-turbo` and default embedding model name is `text-embedding-ada-002`. If you want to use different models from OpenAI, fill in your model names in [src/config.py](./src/config.py).
+{{/useOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT`.
+{{/enableTestToolByDefault}}
+{{^enableTestToolByDefault}}
+{{#useAzureOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure OpenAI key `SECRET_AZURE_OPENAI_API_KEY`, deployment name `AZURE_OPENAI_MODEL_DEPLOYMENT_NAME`, endpoint `AZURE_OPENAI_ENDPOINT` and embedding deployment name `AZURE_OPENAI_EMBEDDING_DEPLOYMENT`.
+{{/useAzureOpenAI}}
+{{#useOpenAI}}
+1. In file *env/.env.local.user*, fill in your OpenAI key `SECRET_OPENAI_API_KEY`. 
+1. In this template, default model name is `gpt-3.5-turbo` and default embedding model name is `text-embedding-ada-002`. If you want to use different models from OpenAI, fill in your model names in [src/config.py](./src/config.py).
+{{/useOpenAI}}
+1. In file *env/.env.local.user*, fill in your Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT`.
+{{/enableTestToolByDefault}}
+
+### Setting up index and documents
+{{^enableTestToolByDefault}}
+1. Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT` are loaded from *env/.env.local.user*. Please make sure you have already configured them.
+{{/enableTestToolByDefault}}
+{{#enableTestToolByDefault}}
+1. Azure Search key `SECRET_AZURE_SEARCH_KEY` and endpoint `AZURE_SEARCH_ENDPOINT` are loaded from *env/.env.testtool.user*. Please make sure you have already configured them.
+{{/enableTestToolByDefault}}
+1. Use command `python -m setup` to create index and upload documents in `src/files`.
```

Comment:
Does `python setup.py` works?
We may also move the markdown files out of `src` folder, as it's not part of the app. Maybe put the markdown files, setup.py and other related files to a folder called `sample_data` or other name you preferred?

Reply:
we put markdown and setup.py in root folder. they are not in src/

Reply:
of course `python setup.py` works. but what is the matter of `python -m setup`? 
In this template, we changed startup command to module mode both local debug and remote deploy. So I think it is better to be aligned for `setup` and `delete` scripts.